packages/amplify-graphql-transformer-interfaces/src/transformer-model-provider.ts

```diff
@@ -0,0 +1,132 @@
+import { ObjectTypeDefinitionNode, DirectiveDefinitionNode, InputValueDefinitionNode } from 'graphql';
+import { TransformerPluginProvider } from './transformer-plugin-provider';
+import { TransformerResolverProvider, TransformerContextProvider, AppSyncDataSourceType, DataSourceInstance } from './transformer-context';
+
+export enum QueryFieldType {
+  GET = 'GET',
+  LIST = 'LIST',
+  SYNC = 'SYNC',
+}
+
+export enum MutationFieldType {
+  CREATE = 'CREATE',
+  UPDATE = 'UPDATE',
+  DELETE = 'DELETE',
+}
+
+export enum SubscriptionFieldType {
+  ON_CREATE = 'ON_CREATE',
+  ON_DELETE = 'ON_DELETE',
+  ON_UPDATE = 'ON_UPDATE',
+}
+export interface TransformerModelProvider extends TransformerPluginProvider {
+  getDataSourceType: () => AppSyncDataSourceType;
+  generateGetResolver: (
+    ctx: TransformerContextProvider,
+    type: ObjectTypeDefinitionNode,
+    typeName: string,
+    fieldName: string,
+    directive?: DirectiveDefinitionNode,
```

Comment:
might be worthwhile to wrap all of these params in one input object and have a type for that so it could be reused in these definitions